### PR TITLE
py-dask: updates all around, add py-dask-mpi, too.

### PR DESCRIPTION
**Pull request by matz-e**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-dask-mpi/package.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    homepage = "https://github.com/dask/dask-mpi"
+    url      = "https://pypi.io/packages/source/d/dask-mpi/dask-mpi-2.0.0.tar.gz"
+
+    import_modules = ['dask_mpi']
+
+    version('2.0.0', sha256='774cd2d69e5f7154e1fa133c22498062edd31507ffa2ea19f4ab4d8975c27bc3')
+
+    depends_on('py-dask@2.2:', type=('build', 'run'))
+    depends_on('py-mpi4py', type=('build', 'run'))
```
Comment (tomdele) [inline var/spack/repos/builtin/packages/py-dask-mpi/package.py:20]: no need of version for py-mpi4py ?

Comment (tomdele) [reply var/spack/repos/builtin/packages/py-dask-mpi/package.py:20]: ok no I checked https://github.com/dask/dask-mpi/blob/master/environment.yml ... 
this is strange tho.

**Review by tomdele — DISMISSED** [review]
I hope the depends_on with all the `when` are correct ... 

Looks good !

**Comment by tomdele** [comment]
Arf, did not refresh and I approved before checking the CI. 

`Error: Package 'py-dask-mpy' not found.`
Not so sure why tho

File: var/spack/repos/builtin/packages/py-dask-mpi/package.py (hunk 1 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/py-dask-mpi/package.py:20]: That environment file is just messed up. There's the `jupyter` reference in there that does not occur anywhere else in the code. Can't say I'm a fan of this way of specifying dependencies that seems to be interwoven with the testing requirements.

**Comment by matz-e** [comment]
> Arf, did not refresh and I approved before checking the CI.
> 
> `Error: Package 'py-dask-mpy' not found.`
> Not so sure why tho

:facepalm: should be fixed now

**Comment by tomdele** [comment]
> :facepalm: should be fixed now

hahah omg I even cut and pasted it without noticing ...

**Review by tomdele — APPROVED** [review]

**Review by pramodk — APPROVED** [review]

**Pull request merged**